mllib/src/main/scala/org/apache/spark/mllib/optimization/LBFGS.scala

```diff
@@ -92,9 +92,11 @@ class LBFGS(private var gradient: Gradient, private var updater: Updater)
   }
 
   /**
+   * :: DeveloperApi ::
    * Set the gradient function (of the loss function of one single data example)
    * to be used for L-BFGS.
    */
+  @DeveloperApi
```

Comment:
Some of its members are public API:
* numCorrections
* convergenceTol
* numIterations
* regParam
* updater

This JIRA/PR could make LBFGS non-DeveloperApi.  Or we could leave this as another to-do item for when we bit the bullet and refactor all the optimization code.